### PR TITLE
[SPARK-47474][CORE] Revert SPARK-47461 and add some comments

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr revert the change of SPARK-47461 and add some comments to `ExecutorAllocationManager#totalRunningTasksPerResourceProfile` to clarify that the tests in `ExecutorAllocationManagerSuite` need to call `listener.totalRunningTasksPerResourceProfile` with `synchronized`.

### Why are the changes needed?
`ExecutorAllocationManagerSuite` need to call `listener.totalRunningTasksPerResourceProfile` with `synchronized`.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala
Comment (LuciferYang) [inline core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala:325]: cc @dongjoon-hyun  @mridulm @tgravescs

Since new comments were added during the revert, I have created a new Jira. Is there a better way to describe the content of the comments?

File: core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala:323]: Let's remove `SPARK-47474: ` here because the comment looks sufficient.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. (one minor comment about comment)

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
BTW, @LuciferYang , as you requested in the above already, I guess we need to wait for @mridulm and @tgravescs 's review and approval. I'll leave this to them.

**Comment by tgravescs** [comment]
+1

**Comment by LuciferYang** [comment]
> BTW, @LuciferYang , as you requested in the above already, I guess we need to wait for @mridulm and @tgravescs 's review and approval. I'll leave this to them.

OK

**Review by mridulm — APPROVED** [review]
Thanks @LuciferYang, @dongjoon-hyun !

**Comment by LuciferYang** [comment]
Merged into master ~ thanks @dongjoon-hyun @tgravescs @mridulm

**Comment by dongjoon-hyun** [comment]
Thank you, @LuciferYang and all!

**Pull request closed without merging**
